src/components/DataTable.tsx

```diff
@@ -160,7 +160,15 @@ export function DataTable<T extends object>(props: Props<T>) {
   }, [perPage, currentPage, filter, sort, status, customFilter]);
 
   const { data, isLoading, isError } = useQuery(
-    [props.endpoint, perPage, currentPage, filter, sort, status, customFilter],
+    [
+      apiEndpoint.pathname,
```

Comment:
This has changed because `props.endpoint` includes the pathname and query parameters (/api/v1/designs?status=active...), and this becomes the first query key while invalidation. This logic forces us to include the whole URL when invalidating queries and unnecessarily complicates the logic. With this change, the first item in the query keys array will be just (/api/v1/designs), allowing us to handle invalidation more easily.